mundane-dsv/src/main/scala/com/ambiata/mundane/dsv/DelimitedValuesParser.scala

```diff
@@ -0,0 +1,68 @@
+package com.ambiata.mundane
+package dsv
+
+import scala.collection.JavaConversions._
+
+import org.apache.commons.csv.{CSVParser, CSVFormat}
+
+import scalaz._, Scalaz._
+
+/**
+ * Create a parser for delimited values based on
+ * different implementations and a choice of
+ *
+ *  - delimiter
+ *  - character used for quoting fields
+ *  - character used to escaped control characters (delimiter, quoteCharacter)
+ *
+ */
+object DelimitedValuesParser {
+
+  def createParser(delimiter: Char, quoteCharacter: Option[Char], escapeCharacter: Option[Char], implementation: DelimitedValuesParserImplementation): DelimitedValuesParser =
+    implementation match {
+      case CommonsCsv => CommonsDelimitedValuesParser(delimiter, quoteCharacter, escapeCharacter)
+    }
+
+  /** use the CommonsCsv implementation as the default implementation */
+  def delimited(delimiter: Char): DelimitedValuesParser =
+    createParser(delimiter, '"'.some, none[Char], CommonsCsv)
+
+  /** use the CommonsCsv implementation as the default implementation */
+  def csv: DelimitedValuesParser =
+    delimited(',')
+
+  /** use the CommonsCsv implementation as the default implementation */
+  def psv: DelimitedValuesParser =
+    delimited('|')
+
+  /** use the CommonsCsv implementation as the default implementation */
+  def tsv: DelimitedValuesParser =
+    delimited('\t')
+}
+
+sealed trait DelimitedValuesParser {
+  def parse(line: String): String \/ List[String]
+}
+
+/**
+ * Implementation of a DelimitedValuesParser based on the Apache Commons CSV library
+ */
+case class CommonsDelimitedValuesParser(delimiter: Char, quoteCharacter: Option[Char], escapeCharacter: Option[Char]) extends DelimitedValuesParser {
+  private[this] val format: CSVFormat = {
+    List(
+      (quoteCharacter, (f: CSVFormat, c: Char) => f withQuote c),
+      (escapeCharacter, (f: CSVFormat, c: Char) => f withEscape c)).foldLeft(CSVFormat.newFormat(delimiter)) {
```

Comment:
Given this is apache commons we are talking about, I have to ask - is CSVFormat thread safe?